src/Rector/Rector/MethodCall/ChangeEntityTraitSetArrayToPatchRector.php

```diff
@@ -0,0 +1,65 @@
+<?php
+declare(strict_types=1);
+
+namespace Cake\Upgrade\Rector\Rector\MethodCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Identifier;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\ConfiguredCodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+final class ChangeEntityTraitSetArrayToPatchRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Replaces $this->set(array) with $this->patch(array) when the first param is an array',
+            [
+                new ConfiguredCodeSample(
+                    <<<'CODE_SAMPLE'
+$this->set(['key' => 'value']);
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+$this->patch(['key' => 'value']);
+CODE_SAMPLE,
+                ),
+            ],
+        );
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class];
+    }
+
+    public function refactor(Node $node): ?Node
+    {
+        if (! $node instanceof MethodCall) {
```

Comment:
Shouldn't there be a check somewhere that the object is instance of `Entity`, so that the change only applies to `Entity::set()` calls only not to `set()` of any other class?

Reply:
Good catch 😁

Reply:
Rector is.... fun to work with. Had to dig around to find a way how to check, if an object reflection uses a trait.